### PR TITLE
feat: show warning errors when table name is not string

### PR DESCRIPTION
**Pull request by MH4GF**
## Description

Even if there are unsupported syntaxes, only warnings are displayed and the system exits normally.

## Background

Previously, if a name was passed to `create_table` as a variable, an error would occur because it was not a string.

```rb
variable = "posts"
create_table variable do |t|
end
```

This does not happen very often because schema.rb is generally generated code, but it often happens when users write their own table definitions while using ActiveRecord's DSL, such as [Ridgepole](https://github.com/ridgepole/ridgepole).

## Solution

Error information is logged but exits normally.
The output is still dirty, but the formatting of the logs will be tackled in a separate issue.

```
node packages/cli/dist-cli/bin/cli.js erd build --input ../../../test/db/Schemafile --format schemarb
UnsupportedTokenError: Expected a string for the table name, but received different data
    at extractTableName (file:///Users/mh4gf/ghq/github.com/liam-hq/liam/frontend/packages/cli/dist-cli/bin/cli.js:21231:20)
    at DBStructureFinder.handleCreateTable (file:///Users/mh4gf/ghq/github.com/liam-hq/liam/frontend/packages/cli/dist-cli/bin/cli.js:21489:28)
    at DBStructureFinder.visitCallNode (file:///Users/mh4gf/ghq/github.com/liam-hq/liam/frontend/packages/cli/dist-cli/bin/cli.js:21531:18)
    at CallNode.accept (file:///Users/mh4gf/ghq/github.com/liam-hq/liam/frontend/packages/cli/dist-cli/bin/cli.js:4330:13)
    at file:///Users/mh4gf/ghq/github.com/liam-hq/liam/frontend/packages/cli/dist-cli/bin/cli.js:612:17
    at Array.forEach (<anonymous>)
    at DBStructureFinder.visitChildNodes (file:///Users/mh4gf/ghq/github.com/liam-hq/liam/frontend/packages/cli/dist-cli/bin/cli.js:611:30)
    at DBStructureFinder.visitStatementsNode (file:///Users/mh4gf/ghq/github.com/liam-hq/liam/frontend/packages/cli/dist-cli/bin/cli.js:1897:10)
    at StatementsNode.accept (file:///Users/mh4gf/ghq/github.com/liam-hq/liam/frontend/packages/cli/dist-cli/bin/cli.js:17967:13)
    at file:///Users/mh4gf/ghq/github.com/liam-hq/liam/frontend/packages/cli/dist-cli/bin/cli.js:612:17
```

### CODE REVIEW

File: frontend/packages/cli/src/cli/runPreprocess.ts [hunk 1]
```diff
@@ -26,7 +26,13 @@ export async function runPreprocess(
...
   let json = null
   try {
-    json = await parse(input, format)
+    const { value, errors } = await parse(input, format)
+    json = value
+    if (errors.length > 0) {
+      for (const error of errors) {
+        console.error(error)
+      }
+    }
```
Comment (MH4GF) [inline frontend/packages/cli/src/cli/runPreprocess.ts:35]: Eventually, we would like to remove try-catch altogether, but this will be addressed in a separate PR.

File: frontend/packages/db-structure/src/parser/schemarb/parser.ts [hunk 2]
```diff
@@ -26,17 +27,34 @@ import type {
...
+
+function extractTableName(
+  argNodes: Node[],
+): Result<string, UnsupportedTokenError> {
   const nameNode = argNodes.find((node) => node instanceof StringNode)
   if (!nameNode) {
-    throw new Error('Table name not found')
+    return err(
+      new UnsupportedTokenError(
+        'Expected a string for the table name, but received different data',
```
Comment (sasamuku) [inline frontend/packages/db-structure/src/parser/schemarb/parser.ts:49]: Table name could be symbol (and we need to support it)

```suggestion
        'Expected string or symbol for the table name, but received different data',
```

**Review by sasamuku — APPROVED** [review]
LGTM 🚀🚀

File: frontend/packages/db-structure/src/parser/schemarb/parser.ts (hunk 2 above)
Comment (MH4GF) [reply frontend/packages/db-structure/src/parser/schemarb/parser.ts:49]: I see, but it just seems that the timing of the message change would be better to support.

Comment (sasamuku) [reply frontend/packages/db-structure/src/parser/schemarb/parser.ts:49]: Absolutely!
